### PR TITLE
Fix diagnostics getting out of sync when having diagnostics_delay_ms set

### LINKED ISSUE #1198: Diagnostics get out of sync
During editing of JSON file the diagnostics become out of sync with the real state of the file. Happened before in other servers and files also.

```
	"diagnostics_delay_ms": 2000,
	"diagnostics_additional_delay_auto_complete_ms": 1000,
```
`st4000-exploration` (b1a1fa79a71d215dde2d6c396b559561feaeef5f).

```
::  -> LSP-json textDocument/didChange: {'textDocument': {'uri': 'file:///project/template/.lokalise.json', 'version': 6}, 'contentChanges': [{'text': '        "filter_langs": ["bg", "de", "en", "es", "fr", "it", "no", "pl", "pt", "pt_BR"],\n', 'range': {'end': {'character': 0, 'line': 7}, 'start': {'character': 0, 'line': 7}}, 'rangeLength': 0}]}
:: <-  LSP-json textDocument/publishDiagnostics: {'diagnostics': [{'message': 'Duplicate object key', 'source': 'json', 'code': 520, 'range': {'end': {'character': 22, 'line': 8}, 'start': {'character': 8, 'line': 8}}, 'severity': 2}, {'message': 'Duplicate object key', 'source': 'json', 'code': 520, 'range': {'end': {'character': 22, 'line': 7}, 'start': {'character': 8, 'line': 7}}, 'severity': 2}], 'uri': 'file:///project/template/.lokalise.json'}
::  -> LSP-json textDocument/didChange: {'textDocument': {'uri': 'file:///project/template/.lokalise.json', 'version': 7}, 'contentChanges': [{'text': '', 'range': {'end': {'character': 0, 'line': 9}, 'start': {'character': 0, 'line': 8}}, 'rangeLength': 86}]}
:: --> LSP-json textDocument/documentColor(10): {'textDocument': {'uri': 'file:///project/template/.lokalise.json'}}
:: <<< LSP-json 10: []
:: <-  LSP-json textDocument/publishDiagnostics: {'diagnostics': [], 'uri': 'file:///project/template/.lokalise.json'}
```

Screenshot 2020-07-20 at 12 53 10

As clear from the log, the last state from the server reported no diagnostics but ST is still showing the one that was reported before that.

### PR DESCRIPTION
**Pull request by rchl**
Resolves #1198

### CODE REVIEW

**Comment by rchl** [comment]
I still see some syncing issue. Will investigate

File: plugin/session_buffer.py [hunk 1]
```diff
@@ -231,7 +235,7 @@ def present() -> None:
             if delay_in_seconds <= 0.0:
                 present()
```
Comment (rwols) [inline plugin/session_buffer.py:236]: Shouldn't we also cancel here?

Comment (rchl) [reply plugin/session_buffer.py:236]: We cancel earlier for all the branches so no? Or what do you mean? :)

Comment (rwols) [reply plugin/session_buffer.py:236]: You're right, I did not notice the above cancellation above.

**Comment by rchl** [comment]
This fixes the original issue I was having but there is another one as mentioned in #1198.
I think it would be fine to merge this already anyway and work on another issue in separate PR.

**Pull request merged**